### DISCUSSION TITLE
Firebase local emulator?

### DISCUSSION THREAD

**Discussion by johngithubby** [firebase/flutterfire#3998] [Q&A] [answered]
Can the Firebase local emulator work with Flutter? I was trying to use the CLI to initialize the project directory as a Firebase project (`firebase init`) but it fails.

**Comment by Salakar** [comment, marked as answer]
Yes it works, see the emulator sections of the documentation for Cloud Firestore & Cloud Functions.

**Reply by curiosbasant** [reply]
it doesn't work with the real device (connected with usb) though?? :(
